tool/teleport/common/teleport.go

```diff
@@ -483,6 +483,13 @@ func onConfigDump(flags dumpFlags) error {
 		return trace.Wrap(err)
 	}
 
+	entries, err := os.ReadDir(flags.DataDir)
+	if !os.IsNotExist(err) && len(entries) != 0 {
+		fmt.Printf(
```

Comment:
1. You probably want to print this to stderr, because people will often `teleport configure > teleport.yaml`.
2. Let's get @xinding33 to review the actual message.

Reply:
Thanks, added in https://github.com/gravitational/teleport/pull/11766/commits/284caff46c62075cfad62bee9d6213bc6d5293c5

```bash
$ ./teleport configure  > teleport.yaml
The data directory /var/lib/teleport has existing cluster state. Running this configuration is likely a mistake. To join a new cluster, specify an alternate --data-dir or clear the /var/lib/teleport directory.

$ cat teleport.yaml
#
# A Sample Teleport configuration file.
# Creates a single proxy, auth and node server.
#
# Things to update:
#  1. license.pem: You only need a license from https://dashboard.goteleport.com
#     if you are an Enterprise customer.
#
version: v2
teleport:
  nodename: work
  data_dir: /var/lib/teleport
  log:
    output: stderr
    severity: INFO
    format:
      output: text
  ca_pin: []
  diag_addr: ""
auth_service:
  enabled: "yes"
  listen_addr: 0.0.0.0:3025
  proxy_listener_mode: multiplex
ssh_service:
  enabled: "yes"
  labels:
    env: example
  commands:
  - name: hostname
    command: [hostname]
    period: 1m0s
proxy_service:
  enabled: "yes"
  https_keypairs: []
  acme: {}
```

Reply:
Nit: rewording to be more concise

```suggestion
The data directory /var/lib/teleport already exists. Specify a different data directory with the --data-dir flag or clear the /var/lib/teleport directory.
```

Agreed that most users will want warnings and errors in stdout.

Regarding the piping behavior, I expect that most users would want and expect `teleport configure` to auto generate a file at `/etc/teleport.yaml`. I would say that should be default behavior and we can add a `--output` (`-o` short form if that's available) flag if they want us to output the contents to a specific file. Thoughts?

Reply:
Makes sense to me. There is already a `-o=stdout` for dumping on stdout, we just have to change the default behavior to save it on `/etc/teleport.yaml`

Reply:
Update it looks like we use in multiple places of the docs `teleport configure`, also in some docker files (21 results in the whole project).

Those would need revision if we change the default behavior, I think we should make this change a separate issue.

Reply:
Update: The team decided that printing to stdout is better to make sure users don't override a config file that they may already have. Also, this behavior abides more to the UNIX philosophy.

Markin this as resolved